### PR TITLE
Use scratch buffer for binary query fixup

### PR DESCRIPTION
**Pull request by doitsujin**
Paves the way for a potential predication rework, and potentially improves performance in case the app directly resolves queries into host memory.

### CODE REVIEW

File: libs/vkd3d/shaders/cs_resolve_binary_queries.comp [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
+};
 
 layout(push_constant)
 uniform u_info_t {
-  uint query_offset;
+  query_buffer_t src_queries;
+  query_buffer_t dst_queries;
   uint query_count;
-} u_info;
+};
```
Comment (HansKristian-Work) [inline libs/vkd3d/shaders/cs_resolve_binary_queries.comp:17]: Sure this compiles? push_constant must have an instance name.

File: libs/vkd3d/command.c [hunk 2]
```diff
@@ -6455,16 +6427,28 @@ static void STDMETHODCALLTYPE d3d12_command_list_ResolveQueryData(d3d12_command_
...
-            get_query_stride(type), VK_QUERY_RESULT_64_BIT | VK_QUERY_RESULT_WAIT_BIT));
+            start_index, query_count, scratch.buffer, scratch.offset, stride,
+            VK_QUERY_RESULT_64_BIT | VK_QUERY_RESULT_WAIT_BIT));
 
-    if (type == D3D12_QUERY_TYPE_BINARY_OCCLUSION)
-        d3d12_command_list_resolve_binary_occlusion_queries(list,
-                buffer, aligned_dst_buffer_offset, query_count);
+    if (do_fixup)
+    {
+        d3d12_command_list_resolve_binary_occlusion_queries(list, scratch.va,
```
Comment (HansKristian-Work) [inline libs/vkd3d/command.c:6449]: Probably a FIXME_ONCE here if device doesn't support BDA.

**Review by HansKristian-Work — COMMENTED** [review]
Looks pretty clean, just some nits.

File: libs/vkd3d/shaders/cs_resolve_binary_queries.comp (hunk 1 above)
Comment (doitsujin) [reply libs/vkd3d/shaders/cs_resolve_binary_queries.comp:17]: glslang isn't complaining. Where does it say that push_constant needs a name (and what's the reason for that anyway, given that ubo doesn't need one)?

Comment (HansKristian-Work) [reply libs/vkd3d/shaders/cs_resolve_binary_queries.comp:17]: hm ... ye, Vulkan GLSL spec says it's optional, I can swear it used to be required ... Anyways, guess this is a non-issue then if it works in glslang.

Comment (doitsujin) [reply libs/vkd3d/shaders/cs_resolve_binary_queries.comp:17]: Ah, actually this requirement was removed three years ago, see https://github.com/KhronosGroup/Vulkan-Docs/issues/428.

Comment (HansKristian-Work) [reply libs/vkd3d/shaders/cs_resolve_binary_queries.comp:17]: Ah, ok, so I'm not insane after all :D

File: libs/vkd3d/command.c (hunk 2 above)
Comment (doitsujin) [reply libs/vkd3d/command.c:6454]: added one.

**Review by HansKristian-Work — APPROVED** [review]
LGTM.

**Pull request merged**